File: org.kevoree.modeling.microframework/src/main/java/org/kevoree/modeling/api/extrapolation/ExtrapolationStrategy.java
package org.kevoree.modeling.api.extrapolation;

import org.kevoree.modeling.api.KObject;
import org.kevoree.modeling.api.meta.MetaAttribute;

/**
 * Created by duke on 10/28/14.
 */
public interface ExtrapolationStrategy {

    public Long[] timedDependencies(KObject current);

    public Object extrapolate(KObject current, MetaAttribute attribute, KObject[] dependencies);

}

File: org.kevoree.modeling.microframework.typescript/src/main/ts/org/kevoree/modeling/api/strategy/ExtrapolationStrategy.ts
///<reference path="../KObject.ts"/>
///<reference path="../meta/MetaAttribute.ts"/>

interface ExtrapolationStrategy {

  timedDependencies(current: KObject<any,any>): number[];

  extrapolate(current: KObject<any,any>, attribute: MetaAttribute, dependencies: KObject<any,any>[]): any;

  mutate(current: KObject<any,any>, attribute: MetaAttribute, payload: any, dependencies: KObject<any,any>[]): void;

}

